apps/barrel_store/src/barrel_metrics.erl

```diff
@@ -72,8 +72,10 @@ incr_counter(Val, Entry) when Val > 0 ->
           ok
       end;
     CntRef when Val == 1 ->
+      hooks:run(metrics, [Entry, Val]),
       mzmetrics:incr_resource_counter(CntRef, 0);
     CntRef ->
+      hooks:run(metrics, [Entry, Val]),
```

Comment:
shouldn't it be the name of the metric ? imo having hooks named `metric.<nameofmetric>` would be interresting. Also using the `hooks` app it will mean one function call / hook instead of folding over all.